app/src/main/res/layout/fragment_connect.xml

```diff
@@ -1,21 +1,78 @@
     android:layout_height="match_parent"
     android:id="@+id/fragment_connect" >
 
-    <include layout="@layout/tab_home" />
-    <include layout="@layout/tab_connect_launch" />
+    <include
+        layout="@layout/tab_home"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent" />
+
+    <include
+        layout="@layout/tab_connect_launch"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent" />
 
-    <TextView
+    <androidx.constraintlayout.widget.Guideline
+        android:id="@+id/guideline_horizontal"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:layout_marginHorizontal="@dimen/margin_text"
-        android:gravity="center"
-        android:textSize="@dimen/size_title"
-        android:textStyle="bold"
-        android:text="@string/text_temp_connect" />
+        android:orientation="horizontal"
+        app:layout_constraintGuide_percent="@dimen/guideline_horizontal_tab" />
+
+    <RelativeLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constrainedHeight="true"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/guideline_horizontal">
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_margin="50dp">
+
+            <com.github.dedis.student20_pop.qrcode.CameraPreview
+                android:id="@+id/qr_camera_preview"
+                android:layout_width="wrap_content"
+                android:layout_height="0dp"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintDimensionRatio="3:4"
+                app:layout_constraintLeft_toLeftOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+        </androidx.constraintlayout.widget.ConstraintLayout>
+
+        <LinearLayout
+            android:id="@+id/camera_permission"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:gravity="center"
+            android:orientation="vertical">
+
+            <TextView
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginHorizontal="@dimen/margin_text"
+                android:gravity="center"
+                android:text="@string/allow_camera_text"
+                android:textSize="@dimen/size_body"
+                android:textStyle="bold" />
+
+            <Button
+                android:id="@+id/allow_camera_button"
+                style="@style/Widget.AppCompat.Button"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:background="#03A9F4"
```

Comment:
You can store the colors in the file res/values/colors.xml and then access them using @colors/id